jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.cpp

```diff
@@ -344,8 +347,245 @@ static ecma_completion_value_t
+
+          if (ecma_is_completion_value_empty (exec_result))
+          {
+            /* 8.f.iii.4. */
+            JERRY_ASSERT (ecma_is_value_object (exec_value));
+            ecma_object_t *exec_obj_p = ecma_get_object_from_value (exec_value);
+
+            ECMA_TRY_CATCH (match_string_value,
+                            ecma_op_object_get (exec_obj_p, index_zero_string_p),
+                            exec_result);
+
+            /* 8.f.iii.5. */
+            ecma_property_descriptor_t prop_desc = ecma_make_empty_property_descriptor ();
+            {
+              prop_desc.is_value_defined = true;
+              prop_desc.value = match_string_value;
+
+              prop_desc.is_writable_defined = true;
+              prop_desc.is_writable = true;
+
+              prop_desc.is_enumerable_defined = true;
+              prop_desc.is_enumerable = false;
+
+              prop_desc.is_configurable_defined = true;
+              prop_desc.is_configurable = true;
+            }
+
+            ecma_string_t *current_index_str_p = ecma_new_ecma_string_from_uint32 (n);
+
+            ecma_completion_value_t completion = ecma_op_object_define_own_property (new_array_obj_p,
+                                                                                     current_index_str_p,
+                                                                                     &prop_desc,
+                                                                                     false);
+            JERRY_ASSERT (ecma_is_completion_value_normal_true (completion));
+
+            ecma_deref_ecma_string (current_index_str_p);
+
+            /* 8.f.iii.6. */
+            n++;
+
+            ECMA_FINALIZE (match_string_value);
+          }
+
+          ECMA_FINALIZE (this_index_number);
+
+          ECMA_FINALIZE (this_index_value);
+        }
+
+        ECMA_FINALIZE (exec_value);
+      }
+
+      if (ecma_is_completion_value_empty (exec_result))
+      {
+        if (n == 0)
+        {
+          /* 8.g. */
+          ret_value = ecma_make_simple_completion_value (ECMA_SIMPLE_VALUE_NULL);
+        }
+        else
+        {
+          /* 8.h. */
+          ret_value = ecma_make_normal_completion_value (ecma_copy_value (new_array_value, true));
+        }
+      }
+      else
+      {
+        ret_value = exec_result;
+      }
+
+
+      ECMA_FINALIZE (new_array_value);
+
+      ECMA_FINALIZE (put_value);
+
+      ecma_deref_ecma_string (last_index_string_p);
+      ecma_deref_ecma_string (index_zero_string_p);
+      ecma_dealloc_number (zero_number_p);
+    }
+
+    ECMA_FINALIZE (global_value);
+
+    ECMA_FINALIZE (global_prop_value);
+
+    ecma_deref_ecma_string (global_string_p);
+
+    ecma_free_value (regexp_value, true);
+  }
+
+  ECMA_FINALIZE (this_to_string_value);
+
+  ECMA_FINALIZE (this_check_coercible_value);
+
+  return ret_value;
 } /* ecma_builtin_string_prototype_object_match */
+#endif /* CONFIG_ECMA_COMPACT_PROFILE_DISABLE_REGEXP_BUILTIN */
```

Comment:
Could you, please, change this to `!CONFIG_ECMA_COMPACT_PROFILE_DISABLE_REGEXP_BUILTIN`, as corresponding directive above is `#ifndef` and so condition checked here is "`CONFIG_ECMA_COMPACT_PROFILE_DISABLE_REGEXP_BUILTIN` is not defined"?